### PR TITLE
Tests for updating audio + SQL queries

### PR DESCRIPTION
**Pull request by eajoyce**
Tests pass for updating audio on a post. 
SQL queries added for updating the post with corresponding audio, selecting all posts for a user, selecting a single post

### CODE REVIEW

File: src/thimble_server/data_access_layer/s3/store.clj [hunk 1]
```diff
@@ -0,0 +1,10 @@
+;; TODO: fns to:
+;;             generate safe filename
+;;             save to s3
+;;             take audio and call other fns and return filename [or something else]
+
+;; just returns static string at the moment for testing other logics
+(defn store-audio!
+  "takes audio, stores to s3 returns filename-key"
```
Comment (dyerw) [inline src/thimble_server/data_access_layer/s3/store.clj]: i'd prefer we don't name this something about s3, as something importing this namespace doesn't need to know that detail.

i'd just put this in data_access_layer/file_storage.clj

**Comment by dwesa** [comment]
Yea was iffy about that. Also I don't like having numbers in filenames

File: src/thimble_server/data_access_layer/post.clj [hunk 2]
```diff
@@ -25,3 +39,17 @@
     (if (and (user-exists? poster)
              (= (insert-post! postid poster is-original-int) 1))
         postid nil)))
+
+(defn add-file-to-post!
+  "takes audio and post id
+   calls file handler
+   updates table so post row has filekey in entry
+   @param   audio  [blob]    the audio file
```
Comment (dyerw) [inline src/thimble_server/data_access_layer/post.clj]: blob is not a datatype. this is more likely to be a BufferedInputStream or some similar Java class. even just putting a "?" for now would be fine.

File: src/thimble_server/data_access_layer/s3/store.clj (hunk 1 above)
Comment (eajoyce) [inline src/thimble_server/data_access_layer/s3/store.clj]: Agreed
Fix before or after merge?

File: src/thimble_server/data_access_layer/sql/select/select_post.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+-- selects post information for a given post ID
+SELECT file, poster FROM posts WHERE id = :postid;
```
Comment (dyerw) [inline src/thimble_server/data_access_layer/sql/select/select_post.sql]: get-post function also returns id yet this doesn't select for that

File: src/thimble_server/data_access_layer/sql/update/update_post_audio.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+--updates file name for a given post id
+UPDATE posts
+SET file = :audiokey
+WHERE id = :id;
```
Comment (dyerw) [inline src/thimble_server/data_access_layer/sql/update/update_post_audio.sql:4]: :+1:

File: src/thimble_server/data_access_layer/sql/select/select_post.sql (hunk 3 above)
Comment (eajoyce) [inline src/thimble_server/data_access_layer/sql/select/select_post.sql]: as a fix I think get-post need not return postid
Query can stay the same

File: test/thimble_server/data_access_layer/post_test.clj [hunk 5]
```diff
@@ -5,19 +5,49 @@
...
+    (is (= (add-file-to-post! "someaudio" 2) nil))))
+
+;; if post does not exists and there is no audio, cannot add to table
+(deftest cannot-add-audio-because-post-DNE-and-no-audio-supplied
+  (with-redefs [post-exists?       (fn [& _] false)
+                store-audio!       (fn [& _] "filename")
+                update-post-audio! (fn [& _] 1)]
+    (is (= (add-file-to-post! "someaudio" 2) nil))))
+
+;; QUESTION/TODO: how do we test adding a filename that already exists?
```
Comment (dyerw) [inline test/thimble_server/data_access_layer/post_test.clj]: you're going to need another SQL function to select for posts with that audio hash and swap it out in the test.

File: src/thimble_server/data_access_layer/sql/select/select_post.sql (hunk 3 above)
Comment (dyerw) [inline src/thimble_server/data_access_layer/sql/select/select_post.sql]: i'd rather return too much information than not enough, just also return that column. it doesn't hurt. it opens up more ways to use the function down the road.

File: src/thimble_server/data_access_layer/s3/store.clj (hunk 1 above)
Comment (dyerw) [inline src/thimble_server/data_access_layer/s3/store.clj]: change it now and push to this branch and it'll update here.

File: src/thimble_server/data_access_layer/sql/select/select_post.sql (hunk 3 above)
Comment (eajoyce) [inline src/thimble_server/data_access_layer/sql/select/select_post.sql]: in the future can we adjust the query to select based on different types of parameters? So the same function could select by postid or audio hash?

Maybe these should just be separate

Comment (dyerw) [inline src/thimble_server/data_access_layer/sql/select/select_post.sql]: that's not a bad idea but don't worry about it for now

File: src/thimble_server/data_access_layer/post.clj [hunk 6]
```diff
@@ -25,3 +58,22 @@
...
              (= (insert-post! postid poster is-original-int) 1))
         postid nil)))
+
+(defn add-file-to-post!
+  "takes audio and post id
+   calls file handler
+   updates table so post row has filekey in entry
+   @param   audio  [?]    the audio file
+   @param   postid [integer] post id the audio is for
+   @returns        [integer] 1 or 0 success or failure"
```
Comment (dyerw) [inline src/thimble_server/data_access_layer/post.clj]: use a boolean

Comment (dyerw) [inline src/thimble_server/data_access_layer/post.clj]: also this lies, it returns nil

**Pull request merged**